### PR TITLE
Setting the default value of COOKIE_DOMAIN to null value

### PR DESCRIPTION
**Pull request by Rajinsharwar**
Setting the default value of COOKIE_DOMAIN to a null value

Trac ticket: https://core.trac.wordpress.org/ticket/46550

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

**Review by jrfnl — APPROVED** [review]
LGTM. For the record, this does NOT set the default value to a "null" value, but to an empty string, which is the correct choice as that's also the default value for the optional parameter in PHP itself.
Also, contrary to setting the value to `null`, setting it to an empty string will prevent a "passing null to non-nullable" deprecation notice in PHP 8.1.

Ref:
* https://www.php.net/setcookie

File: src/wp-includes/default-constants.php [hunk 1]
```diff
@@ -327,7 +327,7 @@ function wp_cookie_constants() {
 	 * @since 2.0.0
 	 */
 	if ( ! defined( 'COOKIE_DOMAIN' ) ) {
-		define( 'COOKIE_DOMAIN', false );
+		define( 'COOKIE_DOMAIN', '' );
```
Comment (mukeshpanchal27) [inline src/wp-includes/default-constants.php:335]: Could you document above that the define value has been changed.

File: src/wp-includes/default-constants.php [hunk 2]
```diff
@@ -325,9 +325,10 @@ function wp_cookie_constants() {
 
 	/**
 	 * @since 2.0.0
+  	 * @since 6.4.0 The default value is empty.
```
Comment (mukeshpanchal27) [inline src/wp-includes/default-constants.php]:
```suggestion
	 * @since 2.0.0
	 * @since 6.4.0 The default value is empty.
```

Fix PHPCS.

Comment (jrfnl) [inline src/wp-includes/default-constants.php]: Couple of nitpicks about the documentation:
1. If this is being documented, the docblock should be above the `define()`, not above the `if ()`. You are, after all, documenting the constant, not the condition.
2. If this is being documented, please actually document _what has changed_. "The default value is empty." is ambiguous. What is empty ? What do you mean by default value ? (constants cannot be changed once defined)

Suggestion for the text: "The standard value has changed from `false` to an empty string".

You may also want to add a `@var string` tag to indicate that the constant is expected to contain a string value (and no other type).

File: src/wp-includes/default-constants.php [hunk 3]
```diff
@@ -327,7 +327,12 @@ function wp_cookie_constants() {
 	 * @since 2.0.0
 	 */
 	if ( ! defined( 'COOKIE_DOMAIN' ) ) {
-		define( 'COOKIE_DOMAIN', false );
+		/**
+		 * @since 6.4.0 The default value has changed from false to an empty string.
```
Comment (jrfnl) [inline src/wp-includes/default-constants.php]:
```suggestion
		 * @since 6.5.0 The value has changed from false to an empty string.
```

**Comment by SergeyBiryukov** [comment]
Thanks for the PR! Merged in [r58011](https://core.trac.wordpress.org/changeset/58011).

**Pull request closed without merging**